### DISCUSSION TITLE
Getting node to automatically adjust in size to fit text

### DISCUSSION THREAD

**Discussion by adamfreeplaneuser01** [freeplane/freeplane#1579] [Help]
<img width="301" alt="Screenshot 2023-12-12 131138" src="https://github.com/freeplane/freeplane/assets/147225942/4f44d454-96af-41c3-bafe-d8f2f0a04126">
<img width="190" alt="Screenshot 2023-12-12 131114" src="https://github.com/freeplane/freeplane/assets/147225942/5fca18a2-7fe9-4ab9-ad2c-f28b53a5f43c">

Hey. I searched Freeplane for an option to enable nodes automatically resizing to fit/reveal all text within the node. I have not, however, been able to find such an option. If anyone knows how to do this, I'd appreciate the help

**Comment by i-plasm** [comment]
Hello, try unchecking the "Uniform" option in the tool panel:
![uniform_node](https://github.com/freeplane/freeplane/assets/144830522/09c82d9d-36ed-4d1e-87de-461259d73196)

**Comment by EdoFro** [comment]
@adamfreeplaneuser01 

Open the **Tool-Panel**
```View->Controls->Tool panel (Ctrl+Shift+P) ```

There you can adjust the **Node shape** parameters

some example images (look at the differences between them):

 
![image](https://github.com/freeplane/freeplane/assets/35700575/dc38b6e6-f6ad-4854-ba75-dafa22ec2368)

![image](https://github.com/freeplane/freeplane/assets/35700575/11b0758c-cb62-4b6b-9893-959f5f772230)

![image](https://github.com/freeplane/freeplane/assets/35700575/9d053c20-fb98-4f5e-adf4-2d896898e510)

![image](https://github.com/freeplane/freeplane/assets/35700575/31485586-076d-4599-bfc9-fd138b792a17)

As said by @i-plasm, the uniform option doesn't resize the node to make that the text box height and its width are simmilar. 
You have to set *Max node width* manually to a value that makes that happens.

BR
edo

**Reply by adamfreeplaneuser01** [reply]
@EdoFro @dpolivaev Sorry, but this isn't adding up. I don't see how there can't be an option to automatically have the node resize to fit all the text. I understanding that this can be done manually, but instinctively I would think there would have to be a way for this to be done automatically without me having to readjust the node every time. Thanks!

**Reply by i-plasm** [reply]
Did you try UN-checking the 'uniform' parameter as shown here: https://github.com/freeplane/freeplane/discussions/1579#discussioncomment-8055966?  having it unchecked makes the text to fit the node, without having to modify any other parameter. At least, I have seen the problem you describe occurring only when that option is on.

**Comment by adamfreeplaneuser01** [comment]
> Did you try UN-checking the 'uniform' parameter as shown here: [#1579 (comment)](https://github.com/freeplane/freeplane/discussions/1579#discussioncomment-8055966)? having it unchecked makes the text to fit the node, without having to modify any other parameter. At least, I have seen the problem you describe occurring only when that option is on.

@i-plasm Hi, so I ran a test with "uniform" checked and unchecked.
<img width="844" alt="br1" src="https://github.com/freeplane/freeplane/assets/147225942/a3cc9319-1772-4782-89af-be78532de8fc">
<img width="722" alt="br" src="https://github.com/freeplane/freeplane/assets/147225942/d9b567bf-e92a-445d-abdc-00989354289e">

When uniform is unchecked, all the text is displayed, but it is extremely packed together, not ideal.

**Comment by dpolivaev** [comment]
Uncheck "uniform" and think well about which values to enter in the "Min node width" and "Max node width" fields. When you like the values, apply them to the default style
